### PR TITLE
Fix incorrect links to Jetpack credentials form

### PR DESCRIPTION
**Pull request by rcanepa**
Fixes 1164141197617539-as-1199174045949517

#### Motivation

1. Fix broken redirects. Currently, all of these redirects that we are changing in this PR take users to `wordpress.com/settings/security` instead of to `wordpress.com/settings/jetpack`.
2. Have unique redirects. We are using the same redirect for multiple purposes. In some cases we want to send users to `/settings/security` but in others, we don't want that. Thus, to have more control, we decided that having unique redirects for each piece of the UI would work better.

#### Changes proposed in this Pull Request:

* Fix redirects that are meant to take users to the server credentials form that are currently taking users to the Security section.
* Use a unique redirect for each place in which we want to redirect users to the server credentials form.

#### Jetpack product discussion

P2: p6TEKc-4bd-p2#comment-8702.
Ticket: 1164141197617539-as-1199174045949517.

#### Does this pull request change what data or activity we track or use?

* No, this PR does not change what data is being currently tracked.

#### Testing instructions:

For context, the redirects added in this PR are being worked in D52679-code. We don't need that revision to test this PR, since we only care about our links using the right redirects, not really where those new redirects take you.

You can use a JN site to test this (make sure to activate the Jetpack Beta Tester plugin), or use the one that comes with this repository.

* With your Jetpack site, purchase any paid plan (we want to make sure that your site has both Jetpack Backup and Jetpack Scan).
* Remove your server credentials if you entered them in the past. You can do this in `https://wordpress.com/settings/jetpack/:site`.
* Go to your site's wp-admin.
* Visit the "At a Glance" Jetpack section.
* In the Security section, check the "Enter credentials" link that appears in the **Scan** dash component.
image.
* Verify that the link includes `source=jetpack-scan-dash-credentials`.
* In the Security section, check the "Enter credentials" link that appears in the **Backup** dash component.
image
* Verify that the link includes `source=jetpack-backup-dash-credentials`.
* Go to the Settings > Security section.
* Verify that the link in the "Backups and security scanning" panel includes `source=jetpack-settings-security-credentials`.
image
* Open a new tab, visit wordpress.com, remove the paid plan from your site, and purchase Jetpack Backup Daily. Close the tab.
* Refresh your wp-admin tab.
* Verify that the link in the "Jetpack Backup" panel includes `source=jetpack-settings-security-credentials`.
image

#### Proposed changelog entry for your changes:

* Fix incorrect links to Jetpack credentials form.

### CODE REVIEW

**Review by scottsweb — DISMISSED** [review]
Excellent, thanks for sorting this so quickly.

File: _inc/client/security/backups-scan.jsx
Comment (jeherve) [inline _inc/client/security/backups-scan.jsx:91]: Shouldn't this one be named `calypso-settings-jetpack`, to better match where it lands?

Comment (scottsweb) [reply _inc/client/security/backups-scan.jsx:91]: There was some discussion in Slack last week. I believe the outcome was for us to use keys what reference where a link points to, not where it has come from. cc @leogermani

Comment (jeherve) [reply _inc/client/security/backups-scan.jsx:91]: Right. Since that links goes to Calypso, shouldn't it stay prefixed with Calypso then, and updated to `jetpack` and instead of `security` since it now leads to the new Jetpack tab in Calypso settings.

Comment (rcanepa) [reply _inc/client/security/backups-scan.jsx:91]: I think we meant to name redirects by their source, not their destination as we do now. The problem with using destinations is that we end up using the same redirect for multiple purposes (pieces of UI, routes, etc.), and then when we one to change one of the pieces of UI that point to that destination, we can't do it without changing the rest. 

This is exactly what is happening with `calypso-settings-security`. We can't change its destination to `/settings/jetpack/` because we still want it to point to `/settings/security` in some cases.

Comment (jeherve) [reply _inc/client/security/backups-scan.jsx:91]: That would make sense then 👍

I'll let @leogermani weigh in, he's the Redirect expert :)

Comment (leogermani) [reply _inc/client/security/backups-scan.jsx:91]: Yes, the idea is that the slug represents where the link is, and not where it points to. I've updated the Fielg Guide last week to reflect that.

This vision wasn't clear before and there are many links in Jetpack that don't follow this pattern. But I think it's a good idea to start doing it right now

**Review by leogermani — APPROVED** [review]
Just make sure to create the redirects on the Redirect manager before merging it

**Comment by rcanepa** [comment]
Today I deployed D52679-code and now redirects work as expected.
cc @leogermani @jeherve

**Review by jeherve — APPROVED** [review]
Approving again after a rebase. It should be good to merge now.

**Review by jeherve — APPROVED** [review]
Trying another rebase, Travis is being annoying, it was not triggered before.

**Review by mattgawarecki — DISMISSED** [review]
LGTM!

**Comment by anomiex** [comment]
Travis is failing due to a broken test, see #17867.

**Comment by rcanepa** [comment]
Now that I rebased the PR, all tests passed.

**Review by anomiex — APPROVED** [review]
Already approved before the rebase by other Crew members, re-approving so it can be merged.

**Pull request merged**
